### PR TITLE
1v1 me first to 31 no powerups [Deathmatch Gamemode]

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

Refactors the old shitty deathmatch mode into an epic proper one with respawning and point tracking and other things.

todo:
- [x] scoreboard
- [x] custom spawning logic with the loadout
- [x] the map

## Why / Balance

just a fun thing for lowpop. why not?

## Technical details

mostly some pretty simple gamerule stuff. kill tracking does some stuff but it shouldn't be awful.

## Media

image
image
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Added the Deathmatch game mode! Brawl it out with other players and try and be the first to 31 kills.

### CODE REVIEW

**Comment by JoeHammad1844** [comment]
team deathmatch?

**Comment by Peptide90** [comment]
+1 for TDM. sum the score per team and bingo.

**Comment by Emisse** [comment]
tdm is out of scope for this pr since it will need additional coding for team balancing/friendly fire/etc

**Comment by Peptide90** [comment]
> tdm is out of scope for this pr since it will need additional coding for team balancing/friendly fire/etc

Friendly fire is part of the charm of losing the game when you accidentally blow up your team. As for team balancing we already have that with nukie gamemode. I can see it being a separate PR though I agree.

**Comment by EmoGarbage404** [comment]
Ready for review!

File: Resources/Locale/en-US/game-ticking/game-presets/preset-deathmatch.ftl [hunk 1]
```diff
@@ -1,2 +1,79 @@
...
+death-match-kill-callout-53 = {CAPITALIZE($killer)} landed a random crit on {$victim}!
+death-match-kill-callout-54 = {CAPITALIZE($killer)} taught {$victim} a valuable lesson!
+death-match-kill-callout-55 = {CAPITALIZE($killer)} hit a home run on {$victim}!
+death-match-kill-callout-56 = {CAPITALIZE($killer)} dunked on {$victim}!
+death-match-kill-callout-57 = {CAPITALIZE($killer)} styled on {$victim}!
+death-match-kill-callout-58 = {CAPITALIZE($killer)} said a rude word to {$victim}!
+death-match-kill-callout-59 = {CAPITALIZE($killer)} sent hate-mail to {$victim}!
+death-match-kill-callout-60 = {CAPITALIZE($killer)} tripped {$victim} down the stairs!
+death-match-kill-callout-env-1 = {CAPITALIZE($victim)} lost a point!
+death-match-kill-callout-env-1 = {CAPITALIZE($victim)} got humiliated!
```
Comment (mirrorcult) [inline Resources/Locale/en-US/game-ticking/game-presets/preset-deathmatch.ftl:70]: duplicated

```suggestion
death-match-kill-callout-env-0 = {CAPITALIZE($victim)} lost a point!
death-match-kill-callout-env-1 = {CAPITALIZE($victim)} got humiliated!
```

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**